### PR TITLE
Dev/iqdat conversion scale

### PR DESCRIPTION
**Pull request by mardet987**
This PR fixes a major bug where the iq data array was being populated incorrectly in the conversion from bfiq to iqdat files. This was brought to my attention by Ray Greenwald and @Cooper10 confirmed. 

The iqdat files were being populated with data in dimensions [num_sequences, num_samps, num_arrays, 2(i+q)] but the actual [iqdat format](https://superdarn.github.io/rst/superdarn/src.doc/rfc/0027.html) uses dimensions [num_sequences, num_arrays, num_samps, 2(i+q)]. 

This PR also includes a small feature to allow scaling of iqdat data. Before we scaled the whole +/- 1V range to the maximum int16. Now there is a configurable scaling factor to allow custom scaling within the int16 range. The default is still to scale the data directly to the max int16 (scaling_factor = 1), to allow clear indication of what was sampled on the receiver, but if the precision of int16 is not adequate the scale could be increased to view the low voltages with greater precision. To incorporate this, the values that are scaled outside of the max/min int16 had to be set to the max/min so that they do not roll over (a sacrifice), but we're bound by int16 in iqdat so :shrug: 

@Cooper10 is currently testing this.

### CODE REVIEW

**Comment by egthomas** [comment]
@mardet987 maybe you've already accounted for this, but watch out for the fact that apparently not all radars store the raw I&Q samples using the same convention, eg

https://github.com/SuperDARN/rst/pull/275#issuecomment-592036031

**Comment by Cooper10** [comment]
I have generated IQ plots from a bfiq file that was converted to iqdat. The plots display the same behaviour as other IQ plots generated directly from iqdat files from old style radars. These plots were made with the iq_plot_listen.py script found under the branch iq_plotting_listening.
20200611 1759 52 cly bm0

**Comment by mts299** [comment]
Probably better for @kevinkrieger to code review but if he is busy I can also do it. Also, you have some conflicts with the develop branch?

**Comment by mts299** [comment]
@Cooper10 @kevinkrieger *poke* *poke*

File: pydarn/io/borealis/borealis_convert.py [hunk 1]
```diff
@@ -505,16 +549,21 @@ def __convert_bfiq_record(borealis_slice_id: int,
                 arrays = [this_data[x, i, :]
                           for x in range(this_data.shape[0])]
                 # append
-                reshaped_data.append(np.ravel(np.column_stack(arrays)))
+                reshaped_data.append(np.ravel(arrays))
```
[the remark this answers is not in the document]
Comment (mardet987) [reply pydarn/io/borealis/borealis_convert.py:552]: affirmative!

File: pydarn/io/borealis/borealis_convert.py [hunk 2]
```diff
@@ -151,13 +151,17 @@ class BorealisConvert(BorealisRead):
...
+        The scaling factor the data has been multiplied by before converting
+        to integers for the corresponding dmap format.
     """
 
     __allowed_conversions = {'rawacf': 'rawacf', 'bfiq': 'iqdat'}
 
     def __init__(self, borealis_filename: str, borealis_filetype: str,
                  sdarn_filename: str, borealis_slice_id: int = None,
-                 borealis_file_structure: Union[str, None] = None):
+                 borealis_file_structure: Union[str, None] = None,
```
[the remark this answers is not in the document]
Comment (mardet987) [reply pydarn/io/borealis/borealis_convert.py:163]: yes I learned this from @mts299 !

**Comment by mardet987** [comment]
@kevinkrieger can you merge?

**Pull request merged**